openupgrade_scripts/scripts/project/15.0.1.2/post-migration.py

```diff
@@ -0,0 +1,108 @@
+from openupgradelib import openupgrade
+
+
+def _convert_field_m2o_to_m2m(env):
+    # Convert m2o to m2m in 'project.task'
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["project.task"],
+        "project_task",
+        "user_ids",
+        openupgrade.get_legacy_name("user_id"),
+    )
+
+
+def _create_missing_project_task_stage_personal(env):
+    tasks = env["project.task"].with_context(active_test=False).search([])
+    tasks._populate_missing_personal_stages()
+    # update for 'done' and 'cancelled' tasks
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE project_task_user_rel rel
```

Comment:
Here you must put on the equivalent "Done" and "Canceled" personal stages for each task that is done or cancel in the main stage, and also you need to map to the first `fold` personal stage if the task stage is another one folded.

Reply:
It's already ok.

![Selection_508](https://user-images.githubusercontent.com/25005517/182659018-4df3246a-344c-4adc-b640-71f122560189.png)

Odoo by default have:
The non-personal tasks (without user_id) are New, In progress, Done and Cancelled.
The personal ones (with user_id) are Inbox, Today, This Week, This Month, Later, Done and Cancelled.

The name is what defines the stage, and the user_id defines if it's personal or not. So the highlighted red part assures to find the right one. In this case, by default, only Done and Cancelled ones.

About the folding, there is nothing to do about. The ones "folded" are Done and Cancelled, so it's already handled as said above.